### ISSUE TITLE
"Killed: 9" while Training NER Pipeline

### ISSUE THREAD

**Issue by kaushik88** [explosion/spaCy#3624]
## How to reproduce the behaviour
I'm trying to train the NER pipeline using cli by running this command but the process was killed (possible due too OOM). However, I even tried this on a 250 GB RAM and even then I encountered the same issue.

python -m spacy train en model training-file.json dev-file.json -b 'en_core_web_md' --pipeline 'ner' --verbose

The size of my training file is around 250 MB and the size of my dev file is around 36 MB. 

Things I've tried - 
1. Tried using the verbose option.
2. Tried debug-data and all the 9 tests pass.
3. Wanted to convert to jsonl but ran into this [issue](https://github.com/explosion/spaCy/issues/3523).
4. Tried reducing the batch_size (but i dont think this is an issue).
5. Tried lowering the num_examples to even 100 but still seeing the issue.

## Debug Data Output

=========================== Data format validation ===========================
✔ Loaded onto.train.json
✔ Loaded onto.development.json
✔ Training data JSON format is valid
✔ Development data JSON format is valid
✔ Corpus is loadable

=============================== Training stats ===============================
Training pipeline: ner
Starting with base model 'en_core_web_md'
1 training docs
1 evaluation docs
✔ No overlap between training and evaluation data

============================== Vocab & Vectors ==============================
ℹ 2200865 total words in the data (54224 unique)
10 most common words: 'the' (116895), ',' (96511), '.' (91639), 'to' (54123),
'of' (52210), 'and' (49793), 'a' (37726), 'in' (35900), 'that' (25649), 'is'
(20239)
ℹ 20000 vectors (684830 unique keys, 300 dimensions)

========================== Named Entity Recognition ==========================
ℹ 0 new labels, 18 existing labels
0 missing values (tokens with '-' label)
Existing: 'LANGUAGE', 'MONEY', 'CARDINAL', 'PRODUCT', 'QUANTITY', 'LAW', 'DATE',
'PERSON', 'WORK_OF_ART', 'TIME', 'GPE', 'EVENT', 'NORP', 'LOC', 'ORDINAL',
'PERCENT', 'FAC', 'ORG'
✔ Good amount of examples for all labels
✔ Examples without occurences available for all labels
✔ No entities consisting of or starting/ending with whitespace

================================== Summary ==================================
✔ 9 checks passed

## Your Environment
* **spaCy version:** 2.1.0
* **Platform:** Darwin-18.2.0-x86_64-i386-64bit
* **Python version:** 3.6.5
* **Models:** en_core_web_md, de_core_news_sm

**Comment by kaushik88** [comment]
I realized it treated the entire json file as 1 document. I split this into multiple documents and it worked.
